termbox/src/lib.rs

```diff
@@ -405,6 +383,55 @@ impl Termbox {
             .unwrap();
         }
     }
+
+    fn send_char(&mut self, to_x: u16, to_y: u16, ch: char) {
+        // if the target cell is next to the last cell, then don't move the cursor
+        if (self.last_cursor.0 + 1) == to_x && self.last_cursor.1 == to_y {
+            self.output_buffer.push(ch as u8);
+        } else {
+            goto(to_x, to_y, &mut self.output_buffer);
+            self.output_buffer.push(ch as u8);
+        }
+        self.last_cursor = (to_x, to_y);
+    }
+}
+
+fn num_to_buf(mut num: u16, buf: &mut Vec<u8>) {
+    let start_len = buf.len();
+    let mut chars_len = 0;
+    loop {
+        let rem = num % 10;
+        let ch = b'0' + rem as u8;
+        buf.push(ch);
+        num /= 10;
+        chars_len += 1;
+        if num == 0 {
+            break;
+        }
+    }
+
+    if chars_len == 1 {
+        return;
+    }
+
+    let swap_len = start_len + (chars_len / 2) as usize;
+    let mut cur_char;
```

Comment:
Can you remove this and just declare it in the next block? (`let cur_char = buf[next_swap_dx]`)